add: レポート編集ページのマークアップ

tkming0916 opened the pull request:
Co-authored-by: sueCalystegia00
# やったこと
- ページ全体のデザインをマークアップに起こした
Figma: https://www.figma.com/file/dfjQLPTZlmvFVPiDT1vFAe/md-report-editor?type=design&node-id=115%3A431&mode=design&t=rE03UNszkh9Cihwb-1

- 大きく「Editor」と「Preview」の2つの要素に分けて，gridで横2つに並べた
- 画像はImageで指定，Iconをimportして利用
- 画面幅に合わせてEditorのツール部分が横スクロールされるようにした


# チェックしてほしいこと
- EditorとPreviewの左右切り替え矢印「⇔」はEditorのヘッダー部分にひっついて動くように実装したつもりだが，他の書き方や適切な書き方があれば教えてほしいです
- Editorの<textarea>が適切に実装できているか（あまり自身ない...）

tkming0916 committed:
add: レポート編集ページのマークアップ

まずはページ全体のデザインをマークアップに起こした。
Figma: https://www.figma.com/file/dfjQLPTZlmvFVPiDT1vFAe/md-report-editor?type=design&node-id=115%3A431&mode=design&t=rE03UNszkh9Cihwb-1

Co-authored-by: sueCalystegia00

sueCalystegia00 commented on next-app/src/app/edit/page.tsx:
```diff
@@ -0,0 +1,253 @@
+              </div>
+
+              {/* レポート入力部分 */}
+              <div className='h-full my-3'>
```
[必須]
子要素側には `flex-grow` をつける

[提案]
親側に `gap` を指定して、こちらから `margin` を外すのはいかが？

```suggestion
              <div className='flex-grow'>
```

tkming0916 replied:
7628bb7704477de5f8cb8c438545e84e157b9b29 で修正しました．

sueCalystegia00 commented on next-app/src/app/edit/page.tsx:
```diff
@@ -0,0 +1,253 @@
+              </div>
+            </div>
+
+            <div className='p-4  h-full'>
```
[必須]
`h-full (height: 100%)` は「親要素の高さの100%」になるため、上部のタブの高さも含んでしまい、現状親要素をはみ出してしまっている

<img width="1822" alt="Screenshot 2024-04-09 at 15 35 20" src="https://github.com/YoslabEngineers/seminar-report-editor/assets/52482280/05149650-fe0a-4606-b78f-df7aba029bb5">

こういうときは、親要素に `flex` を指定し、子要素側に `flex-grow` を指定すると、
子要素を「親要素の残ったスペースいっぱいに広げる」ことができる。

```suggestion
            <div className='flex flex-col gap-3 p-4  flex-grow'>
```

tkming0916 replied:
5ace172d18e84eb06b13a5d68c8cd37c3b4e8789 で修正しました👍

sueCalystegia00 commented on next-app/src/app/edit/page.tsx:
```diff
@@ -0,0 +1,253 @@
+      <div className='w-full h-full p-6'>
+        <div className='w-full h-full grid grid-cols-2 justify-between gap-0 border-2 border-gray-300 rounded-lg overflow-hidden'>
+          {/* エディター */}
+          <div>
```
[必須]

145行目の対応のために `flex` をつける
```suggestion
          <div className='flex'>
```

tkming0916 replied:
5ace172d18e84eb06b13a5d68c8cd37c3b4e8789 で修正しました👍

sueCalystegia00 commented on next-app/src/app/edit/page.tsx:
```diff
@@ -0,0 +1,253 @@
+                </div>
+              </div>
+
+              {/* エディターとプレビューの切り替え部分 */}
+              <div className='absolute top-1/4 right-[-15px]'>
+                <p className='text-3xl'>⇔</p>
+              </div>
```
[提案]
見た目には問題ないけど、HTMLの構造として考えるとここに記述されてるのは気持ち悪い気がする。
(「エディターとプレビューの入れ替えボタン」と「エディターのツール」が並列関係なのか？)

「エディター全体」と「プレビュー全体」の２つの `div` と並列の階層にあるといい気がする

tkming0916 replied:
relativeをeditor側につけるのか，ページ全体のmainにつけるのか，どっちが良きですかね？

sueCalystegia00 replied:
今回切り替えボタンを「エディターとプレビューを包含する要素」の直下に階層に配置するなら、その「包含する要素」にrelativeを付けるべきじゃないかな？
切り替えボタンのabsoluteの基準としたい要素にrelativeを付けるイメージ

tkming0916 replied:
previewのヘッダーに追従する形なのかなって思ったんですけど，htmlの構造的に気持ち悪いですもんね．．．むじい

sueCalystegia00 replied:
エディターとプレビューが常にラッパー要素のwidth50%なら、切り替えボタンもラッパー要素の真ん中にabsoluteで位置指定すれば追従すると思ってたけど違った？

tkming0916 replied:
90708cde1683c4ab8c143b511d297fe857739fa8 で修正しました．

sueCalystegia00 commented on next-app/src/app/edit/page.tsx:
```diff
@@ -0,0 +1,253 @@
+            </div>
+            <div className='p-4'>
+              <p>1. はじめに</p>
+            </div>
```
[必須]
プレビューはゼミレポの完成形が表示されているべき箇所なので、ゼミレポのヘッダー(タイトルとかの枠)は作っておいてほしい。

sueCalystegia00 replied:
大枠出来ている。previewのレスポンシブ対応はコンポーネントに切り分けた後に対応できればいいので、一旦OK

sueCalystegia00 reviewed (changes requested):
@tkming0916 
全体的によく出来てると思う！
レイアウトが一部崩れてる部分とか修正の提案を書いていったので対応よろしく〜
わからないことはいつでも質問してもらえれば👍

tkming0916 committed:
fix: レポート入力部分が上部タブの高さを含んでいたので，親要素をはみ出さないようにした．

PRのコメントURL
<https://github.com/YoslabEngineers/seminar-report-editor/pull/19#discussion_r1557030532>
<https://github.com/YoslabEngineers/seminar-report-editor/pull/19#discussion_r1557031869>

tkming0916 committed:
add: 子要素にflex-grow追加．
提案の親側gapは<5ace172d>で追加済み

PRのコメントURL
<https://github.com/YoslabEngineers/seminar-report-editor/pull/19#discussion_r1557028136>

tkming0916 committed:
change: 切替ボタンをエディターとプレビューに並列の要素とした．親要素はページの本体（ヘッダーの下部分）にした．

PRのコメントURL
<https://github.com/YoslabEngineers/seminar-report-editor/pull/19#discussion_r1557038027>

tkming0916 committed:
add: ゼミレポのヘッダー追加

PRのコメントURL
<https://github.com/YoslabEngineers/seminar-report-editor/pull/19#discussion_r1557041385>

sueCalystegia00 committed:
update: 各コンポーネントに分離

pageに配置された大枠4つのコンポーネントを切り出した。

sueCalystegia00 committed:
refactor: assetsディレクトリがルーティング対象外であることを明示

_ をつけるとNext.jsのルーティング対象となる。
assetsがルーティング対象になることはないが、一応明示しておく。

tkming0916 committed:
Merge pull request #25 from YoslabEngineers/feat/#10_devide-components

FE コンポーネントに分離

sueCalystegia00 reviewed (approved):
LGTM
